File: src/code-challenges/php/isPalindrome.php

<?php 
function isPalindrome($x) {
  $xToString = strval($x);
  $xToStringReversed = "";
  
  for($i = strlen($xToString) - 1; $i >= 0;$i--){
      $xToStringReversed =  "$xToStringReversed"."$xToString[$i]";
  }
  return $xToStringReversed == $xToString ? 1 : 0;
}
print isPalindrome(-121)
?>

File: src/code-challenges/cpp/isPalindrome.cpp
#include <iostream>

using namespace std;

bool isPalindrome(int x) {
  string xToString = to_string(x);
  string xToStringReversed;
  int xToStringLength = xToString.length();
  for(int i = xToStringLength-1; i >= 0;i--){
    xToStringReversed += xToString.at(i);
  }
  if(xToString == xToStringReversed){
    return true;
  }
  return false; 
}

int main() {
//   isPalindrome(121);
  cout << isPalindrome(-121) << endl;
}